tests/activity.go

```diff
@@ -54,6 +55,142 @@ import (
 	"google.golang.org/protobuf/types/known/durationpb"
 )
 
+type ActivityTestSuite struct {
+	ClientFunctionalSuite
+}
+
+func (s *ActivityTestSuite) TestActivityScheduleToClose_FiredDuringBackoff() {
+	// We have activity that always fails.
+	// We have backoff timers and schedule_to_close activity timeout happens during that backoff timer.
+	// activity will be scheduled twice. After second failure (that should happen at ~4.2 sec) next retry will not
+	// be scheduled because "schedule_to_close" will happen before retry happens
+	initialRetryInterval := time.Second * 2
+	scheduleToCloseTimeout := 3 * time.Second
+	startToCloseTimeout := 1 * time.Second
+
+	activityRetryPolicy := &temporal.RetryPolicy{
+		InitialInterval:    initialRetryInterval,
+		BackoffCoefficient: 1,
+		MaximumInterval:    time.Second * 10,
+		MaximumAttempts:    10,
```

Comment:
Don't set maximum attempts, you want to make sure that schedule-to-close is enforced.